create.yml

```diff
@@ -14,33 +15,137 @@
     - name: Configure VPC
       include_role:
         name: configure_vpc
+      vars:
+        name_prefix: "{{ ibmcloud_vpc_name_prefix }}"
+        ibmcloud_ic_region: "{{ ibmcloud_vpc_region }}"
 
     - name: Configure VPC Subnet
       include_role:
         name: configure_vpc_subnet
+      vars:
+        name_prefix: "{{ ibmcloud_vpc_name_prefix }}"
+        ibmcloud_ic_region: "{{ ibmcloud_vpc_region }}"
 
     - name: Configure SSH Key
       include_role:
         name: configure_ssh
+      vars:
+        name_prefix: "{{ ibmcloud_vpc_name_prefix }}"
+        ibmcloud_ic_region: "{{ ibmcloud_vpc_region }}"
 
-    - name: Install the VM's
+    - name: Install the VM's for Tower
       include_tasks: create_vsi.yml
       loop: "{{ range(1, ic_instance |int + 1, 1)|list }}"
+      vars:
+        name_prefix: "{{ ibmcloud_vpc_name_prefix }}"
+        ibmcloud_ic_region: "{{ ibmcloud_vpc_region }}"
+        vsi_profile: "{{ ibmcloud_vsi_profile }}"
+        image_id: "{{ ibmcloud_vsi_image_id }}"
+        ic_instance: "{{ ibmcloud_vsi_count }}"
+
+    - name: Install the VM's for Database
+      include_tasks: create_vsi.yml
+      loop: "{{ range(ibmcloud_vsi_count|int + 1, ic_instance|int + 1, 1)|list }}"
+      vars:
+        name_prefix: "{{ ibmcloud_vpc_name_prefix }}"
+        ibmcloud_ic_region: "{{ ibmcloud_vpc_region }}"
+        vsi_profile: "{{ ibmcloud_vsi_db_profile }}"
+        image_id: "{{ ibmcloud_vsi_image_id }}"
+        ic_instance: "{{ ibmcloud_vsi_db_count|int + ibmcloud_vsi_count|int }}"
+
+    - name: Save Floating IP's of Tower of Database
+      set_fact:
+        tower_node: "{{ ibmcloud_vsi_node[:ibmcloud_vsi_count|int:] }}"
+        db_node: "{{ [ ibmcloud_vsi_node[-1] ] }}"
+        cacheable: True
 
     - name: Configure Security Group Rule
       include_role:
         name: configure_security_group
+      vars:
+        name_prefix: "{{ ibmcloud_vpc_name_prefix }}"
+        ibmcloud_ic_region: "{{ ibmcloud_vpc_region }}"
 
     - name: Add VM's to Tower and Database group
       include_role:
         name: add_hosts
+      when: install_tower | bool
+
+    - name: Print IBM Cloud Instance Floating IPs
+      debug:
+        msg: 
+          - "IC instance Floating IP: "
+          - "Tower: {{ tower_node }}"
+          - "Database: {{ db_node }}"
+
```

Comment:
Isnt the above change of separating Tower and DB installation interfering with the Previous PR, where we made it generic for users who just want to install VM's on IBM cloud. But now we are addressing them as Tower and Database ?
Also, now if a user who only wants to install VM's on IBM cloud will have to put appropriate values for ibmcloud_vsi_count and ibmcloud_vsi_db_count both ?

Reply:
i was thinking of discussing with you about the version "tagging" since we have added lot of other changes that will not allow us satisfy "users who just want to install VM's on IBM cloud". Also we need to update the Readme specifying others to use one perticular commit. 

https://dev.to/neshaz/a-tutorial-for-tagging-releases-in-git-147e


Please let me know if you think this is a bad/good idea

Reply:
I see, I think we can still accomplish the "users who only wants to install VM's" like we had previously. The DB node count will always be 1, so we dont really need that counter. We can just take 1 counter like previously, which is ic_instance_count. That will be number of VM's and in case of Tower it will be tower instances. Now, when the install_tower is true, we can call both the plays of install VM's and install DB VM. But when install_tower is false we can skip the DB instance VM creation. That way, users will only get the VM's which are mentioned via ic_instance_count. Also since you want to have a ability to have a different instance size for DB, you can take the db_instance_profile like you are doing now and just use that while creating the DB instance. Please let me know if this is confusing, we can discuss offline on a call.

Reply:
yeah this makes sense. Let me create a PR for it.

Reply:
Great ! thanks. Sorry, but did you mean a different PR ? or you will just update this one ?

Reply:
i will update the same PR.

Reply:
I have created this commit - https://github.com/jainnikhil30/ibm_cloud_tower_deploy/pull/9/commits/7cd6146c5a24f9d87c3478b3e580db7b2571b012 to reflect the suggested changes.